.github/workflows/samples-kotlin-server-jdk17.yaml

```diff
@@ -2,26 +2,28 @@ name: Samples Kotlin server (jdk17)
 
 on:
   push:
-    branches:
+    paths:
       - 'samples/server/petstore/kotlin-springboot-3*/**'
-      - 'samples/server/petstore/kotlin-server/javalin/**'
-      - 'samples/server/petstore/kotlin-server/javalin-6/**'
+      - 'samples/server/petstore/kotlin-server/**'
+      - 'samples/server/petstore/kotlin-server-modelMutable/**'
+      - 'samples/server/petstore/kotlin-springboot-*/**'
       # comment out due to gradle build failure
       # - samples/server/petstore/kotlin-spring-default/**
   pull_request:
     paths:
       - 'samples/server/petstore/kotlin-springboot-3*/**'
-      - 'samples/server/petstore/kotlin-server/javalin/**'
-      - 'samples/server/petstore/kotlin-server/javalin-6/**'
+      - 'samples/server/petstore/kotlin-server/**'
+      - 'samples/server/petstore/kotlin-server-modelMutable/**'
+      - 'samples/server/petstore/kotlin-springboot-*/**'
       # comment out due to gradle build failure
       # - samples/server/petstore/kotlin-spring-default/**
 
 env:
-  GRADLE_VERSION: 7.4
+  GRADLE_VERSION: '8.10'
```

Comment:
fyi. 7.x is still supported (Critical Bug and Security Fixes): https://endoflife.date/gradle

i'm ok to move forward with 8.x. let's see if our users cannot migrate to 7.x for any reason.